### PR TITLE
WIP: Code refactor to use the repo's existing object structure

### PR DESCRIPTION
**Pull request by vindard**
An internal PR that I'll be using to compare changes between an `experimentation` branch I'm running and my fork's `master` branch pulled from @freehodl's repo.

The goal of this is to use the work that @freehodl has put in so far in adding Tor as a feature to the launcher, and to help him make the code more in line with the object structure already being used thoughout Pierre's original repo.

---
Once the experimentation is finished here, the idea would be to open a separate branch and write commits in a way that helps to show via the diffs where the older coder is being refactored into new code. After that, it'll be to then do a PR into @freehodl's `master` branch so that the updates would be added to [the existing PR](https://github.com/lightning-power-users/node-launcher/pull/117) that compares his branch to the original **node-launcher** repo's `master` branch.

### CODE REVIEW

File: node_launcher/node_set/tor.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    #BITCOIN_CONF_PATH,
+    #TOR_DATA_PATH,
+    #TOR_TORRC_PATH,
+    #LND_CONF_PATH,
+    #TOR_EXE_PATH,
+)
+
+class Tor(object):
+    file: ConfigurationFile
+    software: TorSoftware
```
Comment (justinmoon) [inline node_launcher/node_set/tor.py]: Is this defined anywhere or just a "stub"?

Comment (vindard) [reply node_launcher/node_set/tor.py]: Just an incomplete stub for now. I'm just using this PR to visually see what changes I've made so far. It's not at all functional as yet and this PR isn't meant to be merged anywhere 🙈

File: node_launcher/node_set/node_set.py [hunk 2]
```diff
@@ -41,6 +41,20 @@ def __init__(self):
...
         self.lnd_client = LndClient(self.lnd)
 
+        file_name = 'torrc'
+        tor_dir_path = TOR_DIR_PATH[OPERATING_SYSTEM]
+        self.tor_configuration_file_path = os.path.join(tor_dir_path, file_name)
+        log.info(
+            'tor_configuration_file_path',
+            tor_configuration_file_path=self.tor_configuration_file_path
+        )
+        self.tor = Tor(
```
Comment (vindard) [inline node_launcher/node_set/node_set.py:54]: @justinmoon here's where that `Tor` class you [had asked](https://github.com/vindard/node-launcher/pull/1#discussion_r266217324) about would get instantiated (just realised I had included it already on reviewing the codebase today)

Whole thing is still WIP though so don't run this branch or anything as yet 😅

**Comment by vindard** [comment]
Refactored commits into the [`conf-refactor`](https://github.com/vindard/node-launcher/tree/conf-refactor) branch from this one. **Branches [identical](https://github.com/vindard/node-launcher/compare/conf-refactor..experimentation) now.**

image

**Comment by vindard** [comment]
> After that, it'll be to then do a PR into @freehodl's `master` branch so that the updates would be added to [the existing PR](https://github.com/lightning-power-users/node-launcher/pull/117) that compares his branch to the original **node-launcher** repo's `master` branch.

**Final task for this PR completed: see freehodl#2**

**Comment by PierreRochard** [comment]
I'm finally ready to focus on integrating tor, sorry for the delay!

**Comment by PierreRochard** [comment]
Is this the pull request I should be looking at? https://github.com/lightning-power-users/node-launcher/pull/117

**Comment by PierreRochard** [comment]
I've opened two pull requests from your branches 

https://github.com/lightning-power-users/node-launcher/pull/289

and 

https://github.com/lightning-power-users/node-launcher/pull/290

I'm happy to review, rebase, and merge either one or both

**Comment by vindard** [comment]
Hey @PierreRochard, ok sweet. I had originally opened my PR to @freehodl's branch so that you would only have to focus on PR [lightning-power-users#117](https://github.com/lightning-power-users/node-launcher/pull/117). @freehodl simply needs to approve PR #2 into his branch for the changes to automatically flow into the existing [#117](https://github.com/lightning-power-users/node-launcher/pull/117) PR to your repo.

If we can get @freehodl to merge my PR, I think this would be the best route to maintain the comment/progress narrative on here. 

---

I'll hit @freehodl up personally to see if he can approve my changes. It would also mean that you (@PierreRochard) can close (without merging) PRs [#289](https://github.com/lightning-power-users/node-launcher/pull/289) & [#290](https://github.com/lightning-power-users/node-launcher/pull/290) on your branch, since [#117](https://github.com/lightning-power-users/node-launcher/pull/117) will be the only one with all changes included once @freehodl approves my PR into his branch.

_In the meantime though, **only** for testing and review purposes, running the code on PR [#289](https://github.com/lightning-power-users/node-launcher/pull/289) should be fine._

**Pull request closed without merging**